src/main/java/redis/clients/jedis/JedisSlotBasedConnectionHandler.java

```diff
@@ -44,9 +43,12 @@ public Jedis getConnection() {
         if (result.equalsIgnoreCase("pong")) return jedis;
 
         pool.returnBrokenResource(jedis);
-      } catch (JedisConnectionException ex) {
+      } catch (JedisException ex) {
         if (jedis != null) {
-          pool.returnBrokenResource(jedis);
+          try {
```

Comment:
> I can say no pros or cons for these two suggested changes, but replacing try-catch with try-finally seems to be a very bad idea: earlier that code catched JedisConnectionException from ping(), now it shall catch any JedisException as well, but you suggest to let that exception be thrown further up instead of catching.


@serge-groshev you're right, we shouldn't remove the catch block as otherwise the exception would be propagated incorrectly. I guess we can refactor the block as follows:

```java
    for (JedisPool pool : pools) {
      Jedis jedis = null;
      try {
        jedis = pool.getResource();

        if (jedis == null) {
          continue;
        }

        String result = jedis.ping();

        if (result.equalsIgnoreCase("pong")) return jedis;
        jedis.close();
      } catch (JedisException je) {
        if (jedis != null) {
          jedis.close();
        }
      }
    }
```
@serge-groshev  @HeartSaVioR WDYT?